test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/concurrent/ConcurrentHashMapTest.scala

```diff
@@ -194,6 +194,17 @@ class ConcurrentHashMapTest extends MapTest {
         assertTrue(s"missing value '$value'", encounteredValues.contains(value))
     }
   }
+
```

Comment:
There should be at least one test case checking the `add` functionality of `ConcurentHashMap.KeySetView`. Notably the following behavior should be tested:

- Adding a new key adds the default value.
- Adding an existing key does *not* alter the value (iff this is the JVM behavior, the doc is IMO vague about it).

Reply:
Hmm, I do not understand.

`KeySetView[K, V]` have only `getMappedValue` to return `V`.
And `KeySetView` made from `ConcurrentHashMap` do not support `add/addAll`.
So, I do not think it is possible to check

* the default value is added when a new key is added
* the value is not altered when an exsting key is added

Reply:
Oh, that is supported if the overload specific to `ConcurrentHashMap` of `keySet` is used: https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ConcurrentHashMap.html#keySet-V-

Reply:
It seems that any addition to the `keySet` always throws Exception.
https://github.com/scala-js/scala-js/pull/4367/commits/d636996881b26218886a3f03952105e710b408a9

Reply:
Ah, do you mean using private constructor `KeySetView(map, defaultValue)` ?

Reply:
Ah, I see.